Source/Particles/Collision/UpdateMomentumPerezElastic.H

```diff
@@ -0,0 +1,202 @@
+#ifndef WARPX_PARTICLES_COLLISION_UPDATE_MOMENTUM_PEREZ_ELASTIC_H_
+#define WARPX_PARTICLES_COLLISION_UPDATE_MOMENTUM_PEREZ_ELASTIC_H_
+
+#include <WarpXConst.H>
+#include <AMReX_Random.H>
+#include <cmath> // isnan() isinf()
+
+/* \brief Update particle velocities according to
+ *        F. Perez et al., Phys. Plasmas 19, 083104 (2012).
+ *        LmdD is max(Debye length, minimal interparticle distance).
+ *        L is the Coulomb log. A fixed L will be used if L > 0,
+ *        otherwise L will be calculated based on the algorithm.
+ *        To see if there are nan or inf updated velocities,
+ *        compile with USE_ASSERTION=TRUE.
+*/
+
+template <typename T_R>
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void UpdateMomentumPerezElastic (
+    T_R& u1x, T_R& u1y, T_R& u1z, T_R& u2x, T_R& u2y, T_R& u2z,
+    T_R const n1, T_R const n2, T_R const n12,
+    T_R const q1, T_R const m1, T_R const w1,
+    T_R const q2, T_R const m2, T_R const w2,
+    T_R const dt, T_R const L,  T_R const lmdD)
+{
+
+    // Temporary variabls
+    T_R buf1; T_R buf2;
+
+    T_R constexpr inv_c2 = 1./(PhysConst::c * PhysConst::c);
+
+    // Compute Lorentz factor gamma
+    T_R g1 = std::sqrt(1.+(u1x*u1x+u1y*u1y+u1z*u1z)*inv_c2);
+    T_R g2 = std::sqrt(1.+(u2x*u2x+u2y*u2y+u2z*u2z)*inv_c2);
+
+    // Compute momenta
+    T_R p1x = u1x * m1;
+    T_R p1y = u1y * m1;
+    T_R p1z = u1z * m1;
+    T_R p2x = u2x * m2;
+    T_R p2y = u2y * m2;
+    T_R p2z = u2z * m2;
+
+    // Compute center-of-mass (COM) velocity and gamma
+    buf1 = m1 * g1;
+    buf2 = m2 * g2 + buf1;
+    T_R const vcx = (p1x+p2x) / buf2;
+    T_R const vcy = (p1y+p2y) / buf2;
+    T_R const vcz = (p1z+p2z) / buf2;
+    T_R const vcm = std::sqrt(vcx*vcx+vcy*vcy+vcz*vcz);
+    T_R const gc  = 1./std::sqrt(1.-vcm*vcm*inv_c2);
+
+    // Avoid 1/0 -> nan
+    if ( vcm == 0. ) { return; }
+
+    // Compute vc dot v1 and v2
+    T_R const vcDv1 = (vcx*u1x + vcy*u1y + vcz*u1z)/g1;
+    T_R const vcDv2 = (vcx*u2x + vcy*u2y + vcz*u2z)/g2;
+
+    // Compute p1 star
+    buf2 = ( (gc-1.)/vcm/vcm*vcDv1 - gc )*buf1;
```

Comment:
Here also, I think that the code would be more readable with more explicit variable names:
```
T_R const lorentz_tansform_factor = ( (gc-T_R(1.))/vcm/vcm*vcDv1 - gc ) * g1 * m1;
```

Reply:
Also, in the comment, you could add:
```
// Compute p1 star (momentum in the center-of-mass frame)
```